### PR TITLE
Release 3.0.0 rc.0

### PR DESCRIPTION
**Pull request by jan--f**
This merges current `main` into `release-3.0` without conflicts (it can fast-forward) and adds a single commit to bump the release version to `v3.0.0-rc.0` and tag it accordingly, see 3f8e019c1a0011f41c8b557ae18804041f3ce219.
I expect the UI builds for the old react-ui to fail, that is expected as discussed in #15223.

### CODE REVIEW

**Comment by jan--f** [comment]
@juliusv @Nexucis Seems like on the release branch we can't do the same as we did in #15223.

File: web/ui/react-app/package.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
     "@lezer/lr": "^1.4.2",
     "@nexucis/fuzzy": "^0.4.1",
     "@nexucis/kvsearch": "^0.8.1",
-    "@prometheus-io/codemirror-promql": "0.54.1",
+    "@prometheus-io/codemirror-promql": "0.300.0-rc.0",
```
Comment (Nexucis) [inline web/ui/react-app/package.json:22]: @jan--f that's the same issue than before. This file shouldn't be updated with the new version :)

**Comment by Nexucis** [comment]
@jan--f is it expected there are that much changes just for a release ? It looks like the branch release is not up to date

**Comment by bboreham** [comment]
Yes the release branch is not up to date.  It was branched to create a beta, then work carried on in main.
Personally I don't think that is a huge deal.

Let me try creating a PR which purely updates `release-3.0`.

**Comment by jan--f** [comment]
Thanks everyone, This is rebase on `release-3.0` and I updated `web/ui/react-app/package[-lock].json` to depend on the `codemirror-promql 0.55.0` module. Iiuc this is the proposed solution to the react builds.

If that is correct I only need to retag this.

**Comment by bboreham** [comment]
> Tags are added.

I think this is a mistake - tags should be added after this PR merges into release-3.0.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 ## unreleased
 
+## 3.0.0-rc.0 / 2024-10-31
```
Comment (bboreham) [inline CHANGELOG.md:5]: Not quite, but we can leave it.

**Review by bboreham — APPROVED** [review]
lgtm

**Comment by jan--f** [comment]
> > Tags are added.
> 
> I think this is a mistake - tags should be added after this PR merges into release-3.0.

The tagged commits are still on my branch, but the prometheus repo knows about them. So merging this will add the tags to the release branch.

**Comment by bboreham** [comment]
Pretty close to previous version; I'd say this is good to go:

image

**Pull request merged**